### ISSUE TITLE
Directional Shadowmap renderer stability issues

### ISSUE THREAD

**Issue by phr00t** [phr00t/FocusEngine#113]
With a very high number of shadow casters and the directional shadow system, sometimes flicker begins to happen where rendering seems to get corrupted randomly... likely due to some multithreading problem. Occasionally, the Vulkan device will be lost during this instability. This very rarely happens even with reasonable shadow caster counts. Problem is greatly exacerbated when using multiple cascades.

**Comment by phr00t** [comment]
Example crash:

![image](https://github.com/phr00t/FocusEngine/assets/5983470/33881e10-dbe4-4203-b796-50d8d695c922)

![image](https://github.com/phr00t/FocusEngine/assets/5983470/b58157b3-99b2-4d10-89d6-45814e244f74)

![image](https://github.com/phr00t/FocusEngine/assets/5983470/d4480014-7ce4-46d8-a9b2-54a17c050651)

**Comment by phr00t** [comment]
This commit: https://github.com/phr00t/FocusEngine/commit/1cba6b654b212ba5350ffb64d97b513cda048e68

... doesn't help flicker, but I'm hoping will help with the crashing, maybe.

**Comment by phr00t** [comment]
I still got a "device lost" crashing error, but this time it was outside of shadow rendering (but same context resetting part of drawing). I'm still unsure why this happens, but at least it is very rare (which also makes it very hard to reproduce and debug).

**Comment by phr00t** [comment]
I fixed it! The flickering was not actually related to shadows at all... shadows just increased the draw count enough to cause a buffer pool to be exhausted. Simply increasing the buffer pool larger solves the problem! I do not notice any significant increase of memory. I'm hoping this also resolves "device lost" Vulkan errors, but I suspect it doesn't. I'm closing this for now as "fixed" and may need to reopen some "device lost" issue later if that problem returns.

Fixed with https://github.com/phr00t/FocusEngine/commit/eb3ee01b72b6e64e1a238a4098d1226e9b9f883a
